x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/package_list_grid.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { lazy, Suspense, useMemo, useCallback, useEffect, useRef, useState } from 'react';
+
+import { EuiButtonGroup, EuiFlexGroup, EuiFlexItem, EuiSkeletonText } from '@elastic/eui';
+import type { AvailablePackagesHookType, IntegrationCardItem } from '@kbn/fleet-plugin/public';
+import { noop } from 'lodash';
+
+import { css } from '@emotion/react';
+import {
+  useStoredIntegrationSearchTerm,
+  useStoredIntegrationTabId,
+} from '../../../../hooks/use_stored_state';
+import { useOnboardingContext } from '../../../onboarding_context';
+import {
+  DEFAULT_TAB,
+  LOADING_SKELETON_HEIGHT,
+  SCROLL_ELEMENT_ID,
+  SEARCH_FILTER_CATEGORIES,
+  WITHOUT_SEARCH_BOX_HEIGHT,
+  WITH_SEARCH_BOX_HEIGHT,
+} from './constants';
+import { INTEGRATION_TABS, INTEGRATION_TABS_BY_ID } from './integration_tabs_configs';
+import { useIntegrationCardList } from './use_integration_card_list';
+import { IntegrationTabId } from './types';
+
+interface WrapperProps {
+  useAvailablePackages: AvailablePackagesHookType;
+}
+
+const isIntegrationTabId = (id: string): id is IntegrationTabId => {
+  return Object.keys(INTEGRATION_TABS_BY_ID).includes(id);
+};
+
+const emptyStateStyles = { paddingTop: '16px' };
+
+export const PackageList = lazy(async () => ({
+  default: await import('@kbn/fleet-plugin/public')
+    .then((module) => module.PackageList())
+    .then((pkg) => pkg.PackageListGrid),
+}));
+
+export const PackageListGrid = React.memo(({ useAvailablePackages }: WrapperProps) => {
+  const { spaceId } = useOnboardingContext();
+  const scrollElement = useRef<HTMLDivElement>(null);
+  const [selectedTabIdFromStorage, setSelectedTabIdToStorage] = useStoredIntegrationTabId(
+    spaceId,
+    DEFAULT_TAB.id
+  );
+  const [toggleIdSelected, setToggleIdSelected] =
+    useState<IntegrationTabId>(selectedTabIdFromStorage);
+  const [searchTermFromStorage, setSearchTermToStorage] = useStoredIntegrationSearchTerm(spaceId);
+  const onTabChange = useCallback(
+    (id: string) => {
+      if (!isIntegrationTabId(id)) {
+        return;
+      }
```

Comment:
```suggestion
    (id: IntegrationTabId) => {
```

Reply:
I have a type error that I can't fix if I change id into type `IntegrationTabId `, so haven't changed this part.
<img width="1052" alt="Screenshot 2024-10-07 at 17 34 31" src="https://github.com/user-attachments/assets/04dc4edd-d085-4141-a27d-ed25c55267f0">

Reply:
Oh okay, you are right, I didn't realize that. Then we can just assume the `EuiButtonGroup` won't return anything that is not in the `INTEGRATION_TABS`:
```suggestion
    (stringId: string) => {
      const id = stringId as IntegrationTabId;
```